prow/test/integration/cmd/fakegcsserver/main.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// fakegcsserver runs the open source GCS emulator from
+// https://github.com/fsouza/fake-gcs-server.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+	"github.com/sirupsen/logrus"
+
+	configflagutil "k8s.io/test-infra/prow/flagutil/config"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+type options struct {
+	// config is the Prow configuration. We need this to read in GCS
+	// configurations under plank's default_decoration_config_entries field set
+	// in the integration test's Prow configuration, because we have to
+	// initialize (create) these buckets before we upload into them with
+	// initupload.
+	config              configflagutil.ConfigOptions
+	emulatorPort        uint
+	emulatorPublicHost  string
+	emulatorStorageRoot string
+}
+
+func (o *options) validate() error {
+	if o.emulatorPort > 65535 {
+		return fmt.Errorf("-emulator-port range (got %d) must be 0 - 65535", o.emulatorPort)
+	}
+
+	return nil
+}
+
+func flagOptions() *options {
+	o := &options{config: configflagutil.ConfigOptions{ConfigPath: "/etc/config/config.yaml"}}
```

Comment:
I'm not familiar with this usage, is this a new way of setting default flag value?

Reply:
I forget where I got this from, but the point is to set `/etc/config/config.yaml` as the default path IIUC.

Reply:
this value is set on the deployment file, so we might not need this default path?

Reply:
sure, this is not harmful so 🤷